plutus-emulator/src/Wallet/Emulator/ChainIndex.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE DerivingStrategies   #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE GADTs                #-}
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE TemplateHaskell      #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans  #-}
+module Wallet.Emulator.ChainIndex where
+
+import           Control.Lens
+import           Control.Monad.Freer
+import           Control.Monad.Freer.State
+import           Control.Monad.Freer.TH
+import           Control.Monad.Freer.Writer
+import           Data.Aeson                 (FromJSON, ToJSON)
+import           Data.Function              ((&))
+import           Data.Text.Prettyprint.Doc
+import           GHC.Generics               (Generic)
+import qualified Wallet.API                 as WAPI
+import           Wallet.Emulator.NodeClient (Notification (..))
+
+import           Ledger.Address             (Address)
+import           Ledger.AddressMap          (AddressMap)
+import qualified Ledger.AddressMap          as AM
+
+data ChainIndexEffect r where
+    StartWatching :: Address -> ChainIndexEffect ()
+    WatchedAddresses :: ChainIndexEffect AM.AddressMap
+    ChainIndexNotify :: Notification -> ChainIndexEffect ()
+makeEffect ''ChainIndexEffect
+
+data ChainIndexEvent =
```

Comment:
Linter says this may be a newtype.

Reply:
The other event types in the emulator are `data` too (I suspect this is because of the purescript bridge?)